beacon-chain/sync/pending_attestations_queue.go

```diff
@@ -75,7 +75,10 @@ func (s *Service) processPendingAtts(ctx context.Context) error {
 			delete(s.blkRootToPendingAtts, bRoot)
 			s.pendingAttsLock.Unlock()
 		} else {
-			if !s.seenPendingBlocks[bRoot] {
+			s.pendingQueueLock.RLock()
+			seen := s.seenPendingBlocks[bRoot]
```

Comment:
if you are going to add a lock you need to lock in a lot of places where `seenPendingBlocks` is accessed